### PR TITLE
[Aljazeera] improve _VALID_URL

### PR DESCRIPTION
**Pull request by steebchen**
### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

Add a new type of link to _VALID_URL with a test. I was not sure whether all those info_dict fields are needed for the second test, but just added them.

### CODE REVIEW

File: youtube_dl/extractor/aljazeera.py [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 
 class AlJazeeraIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?aljazeera\.com/programmes/.*?/(?P<id>[^/]+)\.html'
+    _VALID_URL = r'https?://(?:www\.)?aljazeera\.com/(programmes|video)/.*?/(?P<id>[^/]+)\.html'
```
Comment (dstftw) [inline youtube_dl/extractor/aljazeera.py]: Don't capture.

File: youtube_dl/extractor/aljazeera.py [hunk 2]
```diff
@@ -19,7 +19,20 @@ class AlJazeeraIE(InfoExtractor):
...
+            'ext': 'mp4',
+            'title': 'Sierra Leone: 709-carat diamond to be auctioned off',
+            'description': 'A 709-carat diamond is being auctioned in Sierra Leone\'s capital Freetown.',
+            'uploader_id': '665003303001',
+            'timestamp': 1494496255,
+            'upload_date': '20170511',
+        },
+        'add_ie': ['BrightcoveNew'],
+        'skip': 'Not accessible from Travis CI server',
+    }]
```
Comment (dstftw) [inline youtube_dl/extractor/aljazeera.py]: `only_matching`.

**Comment by dstftw** [comment]
Squash commits.

**Comment by steebchen** [comment]
Fixed the changes & squashed it – however, I am still trying to squash this damn b89b54e commit.. Apart from this, did I do everything correctly? And, just out of curiosity, why can't you just squash all those commits when merging the PR?

**Pull request merged**